docs/guides/key-manager/upgrade-lsp6.md

```diff
@@ -0,0 +1,246 @@
+---
+sidebar_label: 'Upgrade LSP6 Key Manager'
+sidebar_position: 4
+---
+
+import Tabs from '@theme/Tabs';
+import TabItem from '@theme/TabItem';
+
+# Upgrade the LSP6 Key Manager
+
+:::info Requirements
+
+You will need a Universal Profile that you can control via its KeyManager to follow this guide.
+If you don't have a Universal Profile yet, follow our previous guide [**Create a Universal Profile**](../universal-profile/create-profile.md) or look at the [_lsp-factory.js_](../../tools/lsp-factoryjs/deployment/universal-profile.md) docs.
+
+:::
+
+In this guide, we will learn how to upgrade the LSP6 Key Manager of your Universal Profile to the latest version available.
+
+By the end of this guide, you will know how to:
+
+- Deploy a new LSP6 Key Manager with the last updates.
+- Transfer ownership of your Universal Profile through the old LSP6 Key Manager.
+- Accept ownership of your Universal Profile through the new LSP6 Key Manager
+
+## Setup
+
+```shell
+npm install @lukso/lsp-smart-contracts web3
+```
+
+## Step 1 - Set up constants
+
+Create a `constants.js` file with the following constants
+
+- `privateKey`: private key of a controller address
+- `keyManagerAddress`: address of the current LSP6 Key Manager
+- `universalProfileAddress`: address of your Universal Profile
+  This controller must have the CHANGEOWNER permission
+
+After this first step you can save & close `constants.js`.
+
+```js
+export const privateKey = '0x...';
+export const universalProfileAddress = '0x...';
+export const keyManagerAddress = '0x...';
+```
+
+## Step 2 - Integrate the Imports
+
+Create a file with the name `upgarde-lsp6.js` and you should have the following imports on the top of the file.
+
+```js
+import LSP6KeyManager from '@lukso/lsp-smart-contracts/artifacts/LSP6KeyManager.json' assert { type: 'json' };
+import {
+  privateKey,
+  keyManagerAddress,
+  universalProfileAddress,
+} from './constants.js';
+
+import Web3 from 'web3';
+const web3 = new Web3('https://rpc.l14.lukso.network');
+```
+
+## Step 3 - Initialize the controller account
+
+In order to use the controller account you need to initialize it using the `web3.js` library.
+
+```js
+const account = web3.eth.accounts.privateKeyToAccount(privateKey);
+web3.eth.accounts.wallet.add(account);
+```
+
+## Step 4 - Initialize the old LSP6 Key Manager
```

Comment:
I would suggest that we move this step in the Step 6 below.

The initial step is to deploy the new Key Manager on the network. So this contract is now ready to be used and linked to the UP.

So after deploying the new Key Manager contract, we could make two sub-sections under step 6.

```
## Step 6 - Upgrade your Key Manager

### Step 6.1 - Transfer Ownership to your new Key Manager

// We create an instance of the old Key Manager (using this code snippet below). We call it with the `transferOwnership(...)` payload and the address of the new Key Manager we deployed in the previous step.

### Step 6.2 - Accept Ownership from your new Key Manager

// then we create an other instance of the new Key Manager, same passing the address of the new Key Manager we deployed in the previous step. And from this new Key Manager, we send the `acceptOwnership(...)` payload.

## The upgrade has been completed successfully.
```